packages/svg-serializer/index.js

```diff
@@ -1,66 +1,134 @@
+
+## License
+
+Copyright (c) 2018 JSCAD Organization https://github.com/jscad
+
+All code released under MIT license
+
+Notes:
+1) CAG conversion to:
+     SVG GROUP containing a SVG PATH for each CAG outline path 
+2) CSG conversion to:
+     mesh
+3) Path2D conversion to:
+     none
+
+TBD
+1) add Path2D conversion
+*/
+
+const {isCAG} = require('@jscad/csg')
 const stringify = require('onml/lib/stringify')
 
 const mimeType = 'image/svg+xml'
 
-const serialize = function (cagObject, options) {
-  options && options.statusCallback && options.statusCallback({progress: 0})
-  var decimals = 1000
 
-  // mirror the CAG about the X axis in order to generate paths into the POSITIVE direction
-  var plane = new CSG.Plane(CSG.Vector3D.Create(0, 1, 0), 0)
-  var cag = cagObject.transform(CSG.Matrix4x4.mirroring(plane))
+/** Serialize the give objects to SVG format.
+ * @param {Object} [options] - options for serialization
+ * @param {Object|Array} objects - objects to serialize as SVG
+ * @returns {Array} serialized contents, SVG format
+ */
+const serialize = (options, ...objects) => {
+  const defaults = {
+    statusCallback: null,
+    unit: 'mm', // em | ex | px | in | cm | mm | pt | pc
+    decimals: 10000
+  }
+  options = Object.assign({}, defaults, options)
+
+  options.statusCallback && options.statusCallback({progress: 0})
 
-  var bounds = cag.getBounds()
-  var paths = cag.getOutlinePaths()
-  var width = Math.round((bounds[1].x - bounds[0].x) * decimals) / decimals
-  var height = Math.round((bounds[1].y - bounds[0].y) * decimals) / decimals
+  // get the lower and upper bounds of ALL convertable objects
+  var bounds = getBounds(objects)
+
+  var width = 0
+  var height = 0
+  if (bounds) {
+    width = Math.round((bounds[1].x - bounds[0].x) * options.decimals) / options.decimals
+    height = Math.round((bounds[1].y - bounds[0].y) * options.decimals) / options.decimals
+  }
 
   var body = ['svg',
     {
-      width: width + 'mm',
-      height: height + 'mm',
+      width: width + options.unit,
+      height: height + options.unit,
       viewBox: ('0 0 ' + width + ' ' + height),
       version: '1.1',
       baseProfile: 'tiny',
       xmlns: 'http://www.w3.org/2000/svg',
       'xmlns:xlink': 'http://www.w3.org/1999/xlink'
     },
-    PathsToSvg(paths, bounds)
   ]
+  if (bounds) {
+    body = body.concat(convertObjects(objects, bounds, options))
+  }
 
   var svg = `<?xml version="1.0" encoding="UTF-8"?>
 <!-- Generated by OpenJSCAD.org -->
 <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1 Tiny//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11-tiny.dtd">
 ${stringify(body)}`
 
-  options && options.statusCallback && options.statusCallback({progress: 100})
+  options.statusCallback && options.statusCallback({progress: 100})
   return [svg]
 }
 
-const PathsToSvg = function (paths, bounds, options) {
-  // calculate offsets in order to create paths orientated from the 0,0 axis
+const getBounds = (objects) => {
```

Comment:
very good ! 
Also close to what CSG.js v2 will do internally, so this is nice !